src/pyfmi/fmi_algorithm_drivers.py

```diff
@@ -605,6 +596,43 @@ def _set_options(self):
                 else:
                     self.with_jacobian = False
 
+    def _set_absolute_tolerance_options(self):
+        """
+        Sets the absolute tolerance. Must not be called before initialization since it depends
+        on state nominals.
+
+        Assumes initial setup of default atol has been done via previous call to _set_options.
+
+        Will try to auto-update absolute tolerances that depend on state nominals retrieved
+        before initialization.
+        """
+        try:
+            atol = self.solver_options["atol"]
+            preinit_nominals = self.model._preinit_nominal_continuous_states
+            if isinstance(atol, str) and atol == "Default":
+                fnbr, _ = self.model.get_ode_sizes()
+                rtol = self.solver_options["rtol"]
+                if fnbr == 0:
+                    self.solver_options["atol"] = 0.01*rtol
+                else:
+                    self.solver_options["atol"] = 0.01*rtol*self.model.nominal_continuous_states
+            elif isinstance(preinit_nominals, N.ndarray) and (preinit_nominals.size > 0):
+                # Heuristic:
+                # Try to find if atol was specified as "atol = factor * model.nominal_continuous_states",
+                # and if that's the case, recompute atol with nominals from after initialization.
+                factors = atol / preinit_nominals
+                f0 = factors[0]
+                for f in factors:
+                    if abs(f0 - f) > f0 * 1e-6:  # REVIEW: How to do this properly?
+                        # REVIEW: Should this give warning? In Case 2 we mention something about it, but hard to understand.
+                        return
+                # Success.
+                self.solver_options["atol"] = atol * self.model.nominal_continuous_states / preinit_nominals

```

Comment:
I'm a bit confused why this is indented to be outside of the if-statement? Should this not be part of the ifelse block above? It seems like we do this update now even if "Default" was used or am I missing something?

Reply:
I think it is part of the elif block?

Reply:
You are right, it is. I missed one indentation when looking at it from the review pane.